tests/commands/diff_test.py

```diff
@@ -0,0 +1,127 @@
+            "huggingface",
+        ]
+        main()
+        captured = capsys.readouterr()
+        assert (
+            _clean_output(captured.out)
+            == """
+ embeddings.word_embeddings.weight, shape = (250, 20)
+ embeddings.position_embeddings.weight, shape = (512, 20)
+ embeddings.token_type_embeddings.weight, shape = (2, 20)
+ embeddings.LayerNorm.weight, shape = (20,)
+ embeddings.LayerNorm.bias, shape = (20,)
+ encoder.layer.0.attention.self.query.weight, shape = (20, 20)
+ encoder.layer.0.attention.self.query.bias, shape = (20,)
+ encoder.layer.0.attention.self.key.weight, shape = (20, 20)
+ encoder.layer.0.attention.self.key.bias, shape = (20,)
+ encoder.layer.0.attention.self.value.weight, shape = (20, 20)
+ encoder.layer.0.attention.self.value.bias, shape = (20,)
+ encoder.layer.0.attention.output.dense.weight, shape = (20, 20)
+ encoder.layer.0.attention.output.dense.bias, shape = (20,)
+ encoder.layer.0.attention.output.LayerNorm.weight, shape = (20,)
+ encoder.layer.0.attention.output.LayerNorm.bias, shape = (20,)
+ encoder.layer.0.intermediate.dense.weight, shape = (40, 20)
+ encoder.layer.0.intermediate.dense.bias, shape = (40,)
+ encoder.layer.0.output.dense.weight, shape = (20, 40)
+ encoder.layer.0.output.dense.bias, shape = (20,)
+ encoder.layer.0.output.LayerNorm.weight, shape = (20,)
+ encoder.layer.0.output.LayerNorm.bias, shape = (20,)
+ pooler.dense.weight, shape = (20, 20)
+ pooler.dense.bias, shape = (20,)
+        """.strip()
+        )
+
+    def test_diff_correct(self, capsys):
+        class ModelA(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.a = Parameter(torch.tensor([1.0, 0.0, 0.0]))
+                self.b = Parameter(torch.tensor([1.0, 0.0, 0.0]))
+                self.c = Parameter(torch.tensor([1.0, 0.0, 0.0]))
+                self.e = Parameter(torch.tensor([1.0, 0.0, 0.0]))
+
+        class ModelB(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.a = Parameter(torch.tensor([1.0, 0.0, 0.0]))
+                self.b = Parameter(torch.tensor([1.0, 0.0, 0.0, 0.0]))
+                self.d = Parameter(torch.tensor([1.0, 0.0, 0.0]))
+                self.e = Parameter(torch.tensor([1.0, 0.0, 1.0]))
+
+        model_a = ModelA()
+        model_b = ModelB()
+
+        torch.save(model_a.state_dict(), self.TEST_DIR / "checkpoint_a.pt")
+        torch.save(model_b.state_dict(), self.TEST_DIR / "checkpoint_b.pt")
+        sys.argv = [
+            "allennlp",
+            "diff",
+            str(self.TEST_DIR / "checkpoint_a.pt"),
+            str(self.TEST_DIR / "checkpoint_b.pt"),
+        ]
+        main()
+        captured = capsys.readouterr()
+        assert (
+            _clean_output(captured.out)
+            == """
+ a, shape = (3,)
+-b, shape = (3,)
+-c, shape = (3,)
++b, shape = (4,)
++d, shape = (3,)
+!e, shape = (3,), difference = 0.5774
```

Comment:
Could you add a small comment that explains how you computed 0.5774?

Reply:
Done.